Utilities/WearableJSONExporter/marketplace-packager.js

```diff
@@ -103,44 +104,80 @@
                 print("Failed to export json");
             }
         }
-        Entities.deleteEntity(entityIDToExport);                        
     }
 
     function onWebEventReceived(event) {
         if (typeof(event) === "string") {
             event = JSON.parse(event);
         }
+        if (event.type === "move-entity") {
+            var entityIDProperties = Entities.getEntityProperties(event.entityID, 'parentID');
+            if (entityIDProperties.parentID === MyAvatar.sessionUUID) {
+                return; // don't change things that are already attached to us
+            }
+            Entities.editEntity(event.entityID, {
+                "parentID" : MyAvatar.sessionUUID,
+                "parentJointIndex" : MyAvatar.getJointIndex(event.joint),
+                "position" : MyAvatar.getJointPosition(event.joint)
+            });
+        }
         if (event.type === "submit" && event.app === "JSON") {
+            tablet.emitScriptEvent("hide-warning");
             var entityID = event.entityID;
-            var joint = event.joint;
-
+            // Collect the rest of the properties we want
             var newExportProperties = exportProperties;
             var newUserDataProperties = baseUserdata;
-    
-            var properties = Entities.getEntityProperties(entityID, ['modelURL', 'dimensions', 'script']);
-
-            newUserDataProperties.Attachment.joint = joint;
+            // Confirm we have a valid joint 
+            var joint = event.joint;
+            if (joint === null) {
+                tablet.emitScriptEvent("display-warning-joint-selection");
+                return;
+            }
+            if (joint.indexOf("Left") !== -1) {
+                newUserDataProperties.Attachment.joint = "[LR]" + joint.substring(4);
```

Comment:
Could make "Left"  and "Right" into  LEFT/RIGHT_JOINT_PREFIX constants     and use LEFT_JOINT_PREFIX.length  to eliminate this magic number